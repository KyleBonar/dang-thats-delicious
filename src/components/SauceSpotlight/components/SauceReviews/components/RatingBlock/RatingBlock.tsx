import {
  StyledContainer,
  StyledDescription,
  StyledEmptyStar,
  StyledFullStar
} from "./RatingBlockStyles";
import ReactRating from "react-rating";
import * as React from "react";

export interface IRatingBlock {
  txt?: string;
  rating?: number;
  name: string;
}

const RatingBlock = ({ txt, rating, name }: IRatingBlock): JSX.Element => {
  return (
    <>
      <StyledContainer>
        <i>{name}:</i>

        {rating && (
          <>
            <ReactRating
              initialRating={rating}
              readonly={true}
              emptySymbol={<StyledEmptyStar height={20} />}
              fullSymbol={<StyledFullStar height={20} />}
            />{" "}
            <small>({rating.toString()})</small>
          </>
        )}
      </StyledContainer>
      <StyledDescription>{txt}</StyledDescription>
    </>
  );
};

export default RatingBlock;
