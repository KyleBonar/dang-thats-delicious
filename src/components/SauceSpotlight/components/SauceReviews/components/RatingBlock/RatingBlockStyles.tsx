import styled from "../../../../../../theme/styled-components";
import Star from "../../../../../../images/icons/Star";

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: nowrap;
  padding: 0 1em;

  > i {
    padding: 0 0.5em 0 0;
  }
`;
StyledContainer.displayName = "div";

export const StyledDescription = styled.p`
  margin-top: 0px;
  padding: 0 2em;
`;
StyledDescription.displayName = "p";

export const StyledEmptyStar = styled(Star)`
  .border {
    fill: ${props => props.theme.primaryThemeColor};
  }

  .center {
    fill: transparent;
  }
`;
StyledEmptyStar.displayName = "Star";

export const StyledFullStar = styled(Star)`
  .border,
  .center {
    fill: ${props => props.theme.primaryThemeColor};
  }
`;
StyledFullStar.displayName = "Star";
