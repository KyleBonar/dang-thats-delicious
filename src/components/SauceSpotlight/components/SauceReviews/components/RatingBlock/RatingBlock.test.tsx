import * as React from "react";
import * as enzyme from "enzyme";

import RatingBlock, { IRatingBlock } from "./RatingBlock";
import {
  ITERATION_SIZE,
  casual
} from "../../../../../../utils/testUtils/testUtils";

const fakeRatingBlockProps = (): IRatingBlock => {
  return {
    txt: casual.random_element([undefined, casual.string]),
    rating: casual.random_element([undefined, casual.integer(1, 5)]),
    name: casual.string
  };
};

describe("<RatingBlock />", () => {
  let wrappers: any = [];
  let props: IRatingBlock[] = [];

  beforeAll(() => {
    wrappers = new Array(ITERATION_SIZE).fill(null).map(() => {
      const fakeProps = fakeRatingBlockProps();
      props.push(fakeProps);

      return enzyme.shallow(<RatingBlock {...fakeProps} />);
    });
  });

  it("renders", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper.exists()).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper.exists()).toMatchSnapshot();
    });
  });

  it("renders expected name", () => {
    wrappers.forEach((wrapper, ind) => {
      const { name } = props[ind];

      expect(wrapper.text()).toContain(name);
    });
  });

  it("renders ReactRating component if rating exists", () => {
    wrappers.forEach((wrapper, ind) => {
      const { rating } = props[ind];
      if (!rating) return;

      expect(wrapper.find("RatingAPILayer").exists()).toBeTruthy();
    });
  });
});
