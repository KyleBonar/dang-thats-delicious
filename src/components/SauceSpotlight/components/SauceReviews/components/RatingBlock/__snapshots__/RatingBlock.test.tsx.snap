// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingBlock /> matches snapshot 1`] = `true`;

exports[`<RatingBlock /> matches snapshot 2`] = `true`;

exports[`<RatingBlock /> matches snapshot 3`] = `true`;

exports[`<RatingBlock /> matches snapshot 4`] = `true`;

exports[`<RatingBlock /> matches snapshot 5`] = `true`;

exports[`<RatingBlock /> matches snapshot 6`] = `true`;

exports[`<RatingBlock /> matches snapshot 7`] = `true`;

exports[`<RatingBlock /> matches snapshot 8`] = `true`;

exports[`<RatingBlock /> matches snapshot 9`] = `true`;

exports[`<RatingBlock /> matches snapshot 10`] = `true`;

exports[`<RatingBlock /> matches snapshot 11`] = `true`;

exports[`<RatingBlock /> matches snapshot 12`] = `true`;

exports[`<RatingBlock /> matches snapshot 13`] = `true`;

exports[`<RatingBlock /> matches snapshot 14`] = `true`;

exports[`<RatingBlock /> matches snapshot 15`] = `true`;

exports[`<RatingBlock /> matches snapshot 16`] = `true`;

exports[`<RatingBlock /> matches snapshot 17`] = `true`;

exports[`<RatingBlock /> matches snapshot 18`] = `true`;

exports[`<RatingBlock /> matches snapshot 19`] = `true`;

exports[`<RatingBlock /> matches snapshot 20`] = `true`;

exports[`<RatingBlock /> matches snapshot 21`] = `true`;

exports[`<RatingBlock /> matches snapshot 22`] = `true`;

exports[`<RatingBlock /> matches snapshot 23`] = `true`;

exports[`<RatingBlock /> matches snapshot 24`] = `true`;

exports[`<RatingBlock /> matches snapshot 25`] = `true`;

exports[`<RatingBlock /> matches snapshot 26`] = `true`;

exports[`<RatingBlock /> matches snapshot 27`] = `true`;

exports[`<RatingBlock /> matches snapshot 28`] = `true`;

exports[`<RatingBlock /> matches snapshot 29`] = `true`;

exports[`<RatingBlock /> matches snapshot 30`] = `true`;

exports[`<RatingBlock /> matches snapshot 31`] = `true`;

exports[`<RatingBlock /> matches snapshot 32`] = `true`;
